src/dmd/mtype.d

```diff
@@ -3841,6 +3841,19 @@ extern (C++) final class TypeDArray : TypeArray
             {
                 return MATCH.convert;
             }
+            else if (next.ty == Tenum)
+            {
+                EnumDeclaration ed = (cast(TypeEnum)next).sym;
+                if (ed.isSpecial())
+                {
+                    /* Special enums that allow implicit conversions to them. 
+                       In particular allow string -> const(wchar_t)[]
+                     */
+                    TypeBasic tob = next.toBasetype().isTypeBasic();
+                    if (tob)
```

Comment:
you can save a LOC here :
`if (TypeBasic tob = next.toBasetype().isTypeBasic())`